pkg/spec/cluster.go

```diff
@@ -92,9 +93,49 @@ func (c *ClusterSpec) Validate() error {
 	return nil
 }
 
+type ClusterPhase string
+
+const (
+	ClusterPhaseRunning = "Running"
+	ClusterPhaseFailed  = "Failed"
+)
+
+type ClusterCondition struct {
+	Type ClusterConditionType `json:"type"`
+
+	Reason string `json:"reason"`
+
+	TransitionTime time.Time `json:"transitionTime"`
+}
+
+type ClusterConditionType string
+
+const (
+	ClusterConditionReady = "Ready"
+
+	ClusterConditionRecovering = "Recovering"
+
+	ClusterConditionScalingUp   = "ScalingUp"
+	ClusterConditionScalingDown = "ScalingDown"
+
+	ClusterConditionUpgrading = "Upgrading"
+)
+
 type ClusterStatus struct {
+	// Phase is the cluster running phase
+	Phase  ClusterPhase `json:"phase"`
+	Reason string       `json:"reason"`
+
+	// Condition keeps ten most recent cluster conditions
+	Conditions []ClusterCondition `json:"conditions"`
```

Comment:
We want to keep a list of conditions.

When user changes size from 3 to 5.

The cluster will turn from condition Ready to ScalingUp. After the cluster has 5 member, the condition will turn from ScalingUp back to Ready.

This is a common practice in k8s world.